src/Pulsar.Client/Internal/PulsarHttpClient.fs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Pulsar.Client.Internal
+
+open System.Net.Http
+open Pulsar.Client.Api
+open System
+
+//  This class is mainly aimed for http lookup service
+type internal PulsarHttpClient (config: PulsarClientConfiguration) =
+
+    let authenticationDataProvider = config.Authentication.GetAuthData()
+
+    let httpClient = new HttpClient(new SocketsHttpHandler(
+        PooledConnectionLifetime = TimeSpan.FromMinutes(2),
+        AllowAutoRedirect = true
+    ))
+
+    member internal this.GetStreamAsync (requestUri: string) =
+        backgroundTask {
+            if authenticationDataProvider.HasDataForHttp() then
+                let request = new HttpRequestMessage(HttpMethod.Get, requestUri)
+                authenticationDataProvider.GetHttpHeaders() |> Seq.iter (
```

Comment:
Please rewrite this into for loop for readability

Reply:
Fixed